File: libasynDiscordWebHook-Nukkit/src/main/java/com/ialdrich23xx/discord/body/embed/components/Field.java
package com.ialdrich23xx.discord.body.embed.components;

import com.ialdrich23xx.discord.body.embed.base.Name;
import com.ialdrich23xx.discord.body.embed.base.Structure;

import java.util.Map;

public class Field extends Structure implements Name {

    private String value;
    private Boolean inline;

    public Field(String name, String value, Boolean inline) {
        this.setName(name);
        this.value = value;
        this.inline = inline;
    }

    public static Field make(String name, String value, Boolean inline) {
        return new Field(name, value, inline);
    }

    public static Field make(String name, String value) {
        return new Field(name, value, false);
    }

    public Field setValue(String value)
    {
        this.value = value;

        return this;
    }

    public String getValue()
    {
        return this.value;
    }

    public Field setInline(Boolean inLine)
    {
        this.inline = inLine;

        return this;
    }

    public Boolean getInline()
    {
        return this.inline;
    }

    @Override
    public Boolean build() {
        return !this.getValue().isEmpty() && !this.getName().isEmpty();
    }

    @Override
    public Map<String, Object> toArray() {
        Map<String, Object> result = this.nameToArray();

        result.put("value", this.getName());
        result.put("inline", this.getInline());

        return result;
    }

    @Override
    public String toString() {
        return "Field(" + this.nameToString() + ",value=" + this.getValue() + ",inline=" + this.getInline().toString() + ")";
    }
}

File: libasynDiscordWebHook-PM/src/ialdrich23xx/libasynwebhook/discord/body/embed/components/Field.php
<?php

declare(strict_types=1);

namespace ialdrich23xx\libasynwebhook\discord\body\embed\components;

use ialdrich23xx\libasynwebhook\discord\body\embed\base\Name;
use ialdrich23xx\libasynwebhook\discord\body\embed\base\Structure;
use function strlen;

class Field extends Structure
{
    use Name;

    public function __construct(
        string $name,
        private string $value,
        private bool $inline = false
    ){
        $this->setName($name);
    }

    public static function make(string $name, string $value, bool $inline = false): self
    {
        return new self($name, $value, $inline);
    }

    public function setValue(string $value): self
    {
        $this->value = $value;

        return $this;
    }

    public function getValue(): string
    {
        return $this->value;
    }

    public function setInline(bool $inLine): self
    {
        $this->inline = $inLine;

        return $this;
    }

    public function getInline(): bool
    {
        return $this->inline;
    }

    public function build(): bool
    {
        return strlen($this->getName()) !== 0 && strlen($this->getValue()) !== 0;
    }

    public function toArray(): array
    {
        $result = $this->nameToArray();

        $result["value"] = $this->getValue();
        $result["inline"] = $this->getInline();

        return $result;
    }

    public function toString(): string
    {
        return "Field(" . $this->nameToString() . ",value=" . $this->getValue() . ",inline=" . $this->getInLine() . ")";
    }
}